solr/core/src/java/org/apache/solr/cloud/overseer/SliceMutator.java

```diff
@@ -129,13 +122,7 @@ public ZkWriteCommand removeReplica(ClusterState clusterState, ZkNodeProps messa
       newSlices.put(slice.getName(), slice);
     }
 
-
-    if (coll.isPerReplicaState()) {
-      PerReplicaStatesOps replicaOps = PerReplicaStatesOps.deleteReplica(cnn, PerReplicaStates.fetch(coll.getZNode(), zkClient, coll.getPerReplicaStates()) );
-      return new ZkWriteCommand(collection, coll.copyWithSlices(newSlices), replicaOps, true);
-    } else {
-      return new ZkWriteCommand(collection, coll.copyWithSlices(newSlices));
-    }
+    return new ZkWriteCommand(collection, coll.copyWithSlices(newSlices));
```

Comment:
In this case, will data remove the prs state of it?

Reply:
In this case, will data node remove the prs state of it?

Reply:
yes